lib/internal/checkUrl.js

```diff
@@ -66,6 +83,12 @@ function checkUrl(link, baseUrl, options, callback)
 		}
 		
 		linkObj.clean(link);
+		if (cache)
+		{
+			cache.store(link.url.resolved, function(cb){
```

Comment:
Ah, agreed. I hadn't looked at the whole file to see that. That name sounds good